src/Analyzer.Utilities/JsonPathResolver.cs

```diff
@@ -74,20 +75,44 @@ public IEnumerable<IJsonPathResolver> Resolve(string jsonPath)
         /// <returns>An enumerable of resolvers with a scope of a resource of the specified type.</returns>
         public IEnumerable<IJsonPathResolver> ResolveResourceType(string resourceType)
         {
-            string fullPath = currentPath + ".resources[*]";
+            string fullPath = this.currentPath + ".resources[*]";
             if (!resolvedPaths.TryGetValue(fullPath, out var resolvedTokens))
             {
                 var resources = this.currentScope.InsensitiveTokens("resources[*]");
                 resolvedTokens = resources.Select(r => (FieldContent)r).ToList();
                 resolvedPaths[fullPath] = resolvedTokens;
             }
 
+            // When trying to resolve Microsoft.Web/sites/siteextensions, for example, we should consider that siteextensions can be a child resource of Microsoft.Web/sites (a parent of the original resource type)
+            var resourceTypeParents = new List<string> { resourceType };
+            var indexesOfTypesSeparators = Regex.Matches(resourceType, "/").Cast<Match>().Select(m => m.Index).Skip(1);
+            foreach (var indexOfTypeSeparator in indexesOfTypesSeparators)
+            {
+                resourceTypeParents.Add(resourceType[..indexOfTypeSeparator]);
+            }
+            
             foreach (var resource in resolvedTokens)
             {
-                if (string.Equals(resource.Value.InsensitiveToken("type")?.Value<string>(), resourceType, StringComparison.OrdinalIgnoreCase))
+                foreach (var resourceTypeParent in resourceTypeParents)
                 {
-                    yield return new JsonPathResolver(resource.Value, resource.Value.Path, this.resolvedPaths);
+                    if (string.Equals(resource.Value.InsensitiveToken("type")?.Value<string>(), resourceTypeParent, StringComparison.OrdinalIgnoreCase))
+                    {
+                        if (resourceTypeParent == resourceType)
```

Comment:
Instead of treating the actual resource being search for like a parent (including it with the list of parents), can we break this out so the code is a little clearer?

Psuedo-code:
```cs
var resourceTypeParents = new List<string>();
// Populate with only parent types

foreach (var resource in resolvedTokens)
{
    if (resource.Type == resourceType)
    {
        yield return new JsonPathResolver(resource);
    }
    else if (resourceTypeParents.Exists(r => r.Type == resource.Type))
    {
        // This resource is a type-parent of the resource being searched for.
        // Search children of this resource for the one in question.
        var parentScope = new JsonPathResolver(resource);
        foreach (var child in parentScope.ResolveResourceType(resourceType))
        {
            yield return child;
        }
    }
}
```